awx/ui_next/src/components/ResourceAccessList/ResourceAccessList.jsx

```diff
@@ -25,8 +25,51 @@ function ResourceAccessList({ i18n, apiModel, resource }) {
   const [showDeleteModal, setShowDeleteModal] = useState(false);
   const location = useLocation();
 
+  const getOrgRoles = org => {
+    const options = Object.entries(org.summary_fields.object_roles).map(opt => {
+      return [opt[1].id.toString(), opt[1].name];
+    });
+    return options;
+  };
+
+  const getResourceRoles = useCallback((org, details) => {
+    let OrgResourceAdminId = null;
+    Object.entries(org?.summary_fields?.object_roles).forEach(([id, value]) => {
+      if (id.indexOf(details.type) > -1) {
+        OrgResourceAdminId = value.id;
+      }
+    });
+
+    const options = Object.entries(details.summary_fields.object_roles).map(
+      opt => {
+        const orgAdmin = org?.summary_fields?.object_roles?.admin_role.id;
+        const orgExecute = org?.summary_fields?.object_roles?.execute_role.id;
+        if (opt[1].name.includes('Admin')) {
```

Comment:
Matching on strings that come back from the API always makes me a little nervous.  This mostly stems from the fact that strings _could_ be translated.  I'm not sure if these particular strings get translated or not (or whether they could be in the future).  I'm wondering if we could do something like this instead:

```
        if (opt[0] === 'admin_role') {
          return [
            [opt[1].id, orgAdmin, OrgResourceAdminId].join(','),
            opt[1].name,
          ];
        }
        if (opt[0] === 'execute_role') {
          return [
            [opt[1].id, orgAdmin, orgExecute, OrgResourceAdminId].join(','),
            opt[1].name,
          ];
        }
```

I'm having some trouble fully wrapping my head on what's going on here.  I may need you to give me the rundown so that I understand how we're generating these options for the dropdown.

Reply:
OK, I see the data structure now.  We want to build a concatenated list of role id's to associate with the "Admin" or "Execute" options like this:

```
["108,2,9", "Admin"]
```